src/app-host/live-reload-client.js

```diff
@@ -0,0 +1,181 @@
+    ];
+    var pendingFilesToRefresh = {};
+    var stylesheets;
+    var socket = sock;
+
+    /**
+     * Reload the page. Currently, only does a naive window.location.reload().
+     */
+    function reloadPage() {
+        window.location.reload(true);
+    }
+
+    /**
+     * Returns the name of the reference attribute (either "url", "href" or "src") that is defined for the given node. If the node defines more than one, returns the first encountered, in that order.
+     *
+     * @param {Element} domNode The DOM node to check.
+     * @returns {String} "url", "href" or "src", or null if none of these attributes is defined.
+     */
+    function getReferenceAttributeForNode(domNode) {
+        if (domNode.getAttribute(urlAttribName)) {
+            return urlAttribName;
+        }
+
+        if (domNode.getAttribute(hrefAttribName)) {
+            return hrefAttribName;
+        }
+
+        if (domNode.getAttribute(srcAttribName)) {
+            return srcAttribName;
+        }
+
+        return null;
+    }
+
+    /**
+     * Checks whether the given URL corresponds to a given file path from the server.
+     *
+     * @param {String} url The URL to check.
+     * @param {String} fileRelativePath The path of the modified file to check, relative to the webRoot.
+     * @returns {boolean} Whether the URL points to the modified file from the server.
+     */
+    function urlMatchesPath(url, fileRelativePath) {
+        var isMatch = false;
+
+        localUrlPrefixes.find(function (prefix) {
+            isMatch = prefix + fileRelativePath === url;
+
+            return isMatch;
+        });
+
+        return isMatch;
+    }
+
+    /**
+     * Finds all the DOM elements that have a reference attribute ("url", "href" or "src") pointing to the given relative path. Excludes <script> tags. 
+     *
+     * @param {String} fileRelativePath The URL of the file to check, relative to the webRoot.
+     * @returns {{ domNode: Element, referenceAttribute: string }[]} An array of "resources" referencing the given file.
+     */
+    function findDomNodesForFilePath(fileRelativePath) {
+        // To use querySelectorAll to query elements based on their attributes, the selector's syntax is: '[attrib1], [attrib2], ...'.
+        var selectorString = `[${referenceAttributes.join('], [')}]`;
+        var rawNodes = document.querySelectorAll(selectorString);
+        var filteredNodes = [];
+
+        // querySelectorAll() does not return an array, so we can't use Array.prototype.filter().
+        for (var i = 0; i < rawNodes.length; ++i) {
+            // Ignore <script> tags (we need to do a full reload for scripts).
+            if (rawNodes[i].tagName.toLowerCase() === "script") {
+                continue;
+            }
+
+            // Verify if the node is referencing the modified file
+            var referenceAttribute = getReferenceAttributeForNode(rawNodes[i]);
+            var nodeReference = rawNodes[i].getAttribute(referenceAttribute);
```

Comment:
You're using multiple times rawNodes[i]. Maybe you can just save it to a variable, and re-use it.